### PR TITLE
fix(cli): Fixes .env file and process.env support in configure

### PR DESCRIPTION
**Pull request by steve-calvert-glean**
## Description

This pull request introduces several updates to improve the configuration and validation processes for Glean client credentials, enhance debugging capabilities, and expand test coverage. Key changes include refining credential handling logic, updating debugging configurations, and adding new test cases to ensure robust validation.

### Configuration and Credential Handling:
* Refactored `configure` function in `src/configure.ts` to load credentials from multiple sources (flags, environment variables, `.env` files) and prioritize them appropriately. Improved error messages for missing credentials. [[1]](diffhunk://#diff-5e04f9154aa4746216a242b0d59346f84204e10d27ca9d386c6d3bc198b8abdcL168-R173) [[2]](diffhunk://#diff-5e04f9154aa4746216a242b0d59346f84204e10d27ca9d386c6d3bc198b8abdcL181-R191)
* Updated `validateFlags` in `src/index.ts` to account for credentials from all sources, ensuring proper validation and providing warnings or errors when necessary. [[1]](diffhunk://#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80L47-R60) [[2]](diffhunk://#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80L79-R81)

### Debugging Enhancements:
* Modified `.vscode/launch.json` to include the `--no-file-parallelism` flag in test runner arguments for better debugging consistency.

### Test Coverage Improvements:
* Added comprehensive test cases in `src/test/cli.test.ts` to validate various credential configurations, including flags, `.env` files, and environment variables. Tests also ensure proper error handling for incomplete or conflicting inputs.
* Enhanced `validateFlags` tests in `src/test/validate-flags.test.ts` to cover edge cases, such as partial credentials or fallback to environment variables. [[1]](diffhunk://#diff-3d56b01a05c6af8653222045d5d48f22d44cc6c16a7049a40440558b7397666fL33-R36) [[2]](diffhunk://#diff-3d56b01a05c6af8653222045d5d48f22d44cc6c16a7049a40440558b7397666fR132-R252)

These changes collectively improve the robustness and usability of the Glean client configuration process while ensuring better debugging and testing practices.

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Code refactoring
- [ ] Build/CI pipeline changes
- [ ] Other (please describe):

## How Has This Been Tested?

- [x] Unit tests
- [x] Integration tests
- [x] Manual testing
- [ ] Other (please describe):

## Checklist

- [x] My code follows the code style of this project
- [ ] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] I have checked for potential breaking changes and addressed them

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
       "autoAttachChildProcesses": true,
       "skipFiles": ["<node_internals>/**", "**/node_modules/**"],
       "program": "${workspaceRoot}/node_modules/vitest/vitest.mjs",
-      "args": ["run", "${relativeFile}"],
+      "args": ["run", "${relativeFile}", "--no-file-parallelism"],
```
Comment (steve-calvert-glean) [inline .vscode/launch.json:12]: This makes debugging vitest much easier!

**Review by robert-jackson-glean — APPROVED** [review]
